File: artifacts/bsu-tv-hub/android/app/src/main/java/edu/bridgew/tvhub/BootReceiver.java
package edu.bridgew.tvhub;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Listens for BOOT_COMPLETED so the hub launches automatically
 * whenever the Bravia TV powers on.
 */
public class BootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        if (Intent.ACTION_BOOT_COMPLETED.equals(action)
                || "android.intent.action.QUICKBOOT_POWERON".equals(action)) {

            Intent launch = new Intent(context, MainActivity.class);
            launch.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK
                    | Intent.FLAG_ACTIVITY_CLEAR_TOP);
            context.startActivity(launch);
        }
    }
}

File: artifacts/gtv-kiosk/app/src/main/java/edu/bridgew/tvkiosk/BootReceiver.kt
package edu.bridgew.tvkiosk

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * Android TV has no boot-to-URL equivalent to the Sony Pro displays, so the
 * shell launches itself on power-up instead. Close enough to kiosk behaviour
 * without fighting the Google TV launcher for the Home button.
 */
class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action ?: return
        if (action != Intent.ACTION_BOOT_COMPLETED &&
            action != "android.intent.action.LOCKED_BOOT_COMPLETED"
        ) return

        context.startActivity(
            Intent(context, MainActivity::class.java)
                .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        )
    }
}
